src/api_client.rs

```diff
@@ -58,7 +59,12 @@ impl Client {
         path: &str,
     ) -> Result<T, Error> {
         let target = self.endpoint.join(path)?;
+        
+        // add test to make sure url is formatted correctly
+        assert_eq!(&target.to_string(), "http://127.0.0.1:8003/eth/v1/node/version/");
```

Comment:
you could add this somewhere else if you wanted

the idea with `get` here is that any of the other methods for `Client` could call it so we don't want to specialize to just the node version endpoint

Reply:
yea it's just there for sanity checking - I wanted to check that the error didn't result from an incorrectly formatted url